core/src/main/java/com/graphhopper/storage/Graph.java

```diff
@@ -100,16 +101,47 @@
     /**
      * Returns an EdgeExplorer which makes it possible to traverse all filtered edges of a specific
      * node. Calling this method might be expensive, so e.g. create an explorer before a for loop!
-     *
-     * @see EdgeExplorer
-     * @see Graph#createEdgeExplorer()
      */
     EdgeExplorer createEdgeExplorer(EdgeFilter filter);
 
     /**
-     * @see Graph#createEdgeExplorer(com.graphhopper.routing.util.EdgeFilter)
+     * Creates an EdgeExplorer that accepts all edges
+     *
+     * @see #createEdgeExplorer(EdgeFilter)
      */
-    EdgeExplorer createEdgeExplorer();
+    default EdgeExplorer createEdgeExplorer() {
+        return createEdgeExplorer(EdgeFilter.ALL_EDGES);
+    }
+
+    /**
+     * Returns an EdgeExplorer that accepts only outgoing edges with finite weight according to the given weighting.
+     *
+     * @see #createEdgeExplorer(EdgeFilter)
+     */
+    default EdgeExplorer createOutEdgeExplorer(Weighting weighting) {
+        final BooleanEncodedValue accessEnc = weighting.getFlagEncoder().getAccessEnc();
+        return createEdgeExplorer(edge -> {
+            boolean access = edge.get(accessEnc);
+            if (!access)
+                return false;
+            return Double.isFinite(weighting.calcEdgeWeight(edge, false));
```

Comment:
This might make using this explorer too expensive for some cases, because if the access flag is set we would have to calculate the weight (possibly expensive e.g. for custom weighting) even though all we want to know is whether its finite or not. Something like `weighting.isEdgeWeightFinite(edge, false)` could help optimizing this. But probably there is no need to optimize this before there is an actual performance issue (so far this new explorer is not used anywhere).